### ISSUE TITLE
`static` variables do not have `static` modifier

### ISSUE THREAD

**Issue by maxwase** [rust-lang/rust-analyzer#16894]
It seems that after #16860 PR local variables declarations became indistinguishable from static variables declarations.

**rust-analyzer version**:  rust-analyzer version: 0.4.1887-standalone (b91697de8 2024-03-19)

**rustc version**: rustc 1.75.0 (82e1608df 2023-12-21)

**relevant settings**: 
Previously I used the following config, this allowed me to diversify between `static`, `static mut`, `const`, `let` and `let mut`
```json
    "editor.semanticTokenColorCustomizations": {
        "[ThemeName]": {
         "enabled": true,
         "rules": {
             "variable": "#FFFFFF",
              "variable.mutable": {
                  "italic": true,
                  "foreground": "#FFFFFF"
              },
             "*.static": {
                 "bold": true,
                 "foreground": "#cc711d",
             },
             "*.static.mutable": {
                 "bold": true,
                 "foreground": "#cc711d",
             },
             "*.static.constant": {
                 "bold": true,
                 "foreground": "#F0F080",
             },
            }
        }
    }
````

**code snippet to reproduce**: 

(Can be triggered via `Developer : Inspect Editor Tokens and Scopes` with the cursor on a symbol name)
```rust
// semantic token type: variable
// modifiers: declaration
static FOO: AtomicUsize = AtomicUsize::new(0);

// semantic token type: variable
// modifiers: declaration constant
const BAR: usize = 0;

fn test() {
    // semantic token type: variable
    // modifiers: declaration constant
    let foo: usize = 0;
}
```

**Comment by maxwase** [comment]
Perhaps it is a separate bug, but now constants do not have `const` modifier (and statics) in doc-commens as well
![image](https://github.com/rust-lang/rust-analyzer/assets/23321756/b5ac12b4-a1ad-4c4f-a0d9-64b2e3f78027)

**Comment by maxwase** [comment]
Resolved by #17074
